orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionRunnerSupport.java

```diff
@@ -113,13 +105,29 @@ public ExecutionRunnerSupport(Collection<StageDefinitionBuilder> stageDefinition
         branchBuilder.postBranchGraph(stage, postBranch);
       });
       callback.accept(singleton(stage), afterGraph);
+
+      // ensure parallel stages have the correct stage type (ie. createServerGroup -> deploy to satisfy deck)
+      parallelStages.forEach(it -> it.setType(branchBuilder.getType()));
```

Comment:
For BranchingStageDefinitionBuilder, the generated parallel stages may (in the case of deploy) need a different stage type.